event-system/src/subscribe.js

```diff
@@ -1,21 +1,64 @@
+import _ from 'lodash';
 import express from 'express';
+import redis from './redis.js';
 
 let router = express.Router();
 
-/*
- * 404 NOT FOUND {error: 'Unknown namespace/room'}
- */
+router.get('/subscribe/:namespace/:room', (req, res) => {
+  let client = redis.createClient();
 
-router.get('/subscriptions/:namespace/:room', (req, res) => {
-  res.send();
-});
+  res.set('Content-Type', 'text/event-stream')
+    .set('Cache-Control', 'no-cache')
+    .set('Connection', 'keep-alive')
+    .write('\n');
 
-router.delete('/subscriptions/:namespace/:room', (req, res) => {
-  res.send();
-});
+  let subscriber = redis.createClient();
+  let channel = req.params.namespace + ':' + req.params.room;
+
+  subscriber.subscribe(channel);
+
+  subscriber.on('message', (channel, message) => {
+    let event = JSON.parse(message);
+    if (isAuthorized(event.acl)) {
+      sendEvent(event.id, event.event, event.data);
+    }
+  });
+
+  let lastEventId = parseInt(req.get('Last-Event-ID'));
+  if (lastEventId) {
+    client.get('current-event-id').then((value) => {
+      client.mget.apply(client, _.map(new Array(value - lastEventId + 1), (elem, i) => {
+        return lastEventId + i;
+      })).then((values) => {
+        return _.map(values, JSON.parse);
+      }).then((events) => {
+        return _.filter(events, (event) => {
+          if (event) {
+            return event.channel === channel && isAuthorized(event.acl);
+          }
+        });
+      }).then((events) => {
+        _.each(events, (event) => {
+          sendEvent(event.id, event.event, event.data);
+        });
+      });
+    });
+  }
+
+  let isAuthorized = (acl) => {
+    return _.indexOf(acl, '*') + 1 || _.indexOf(acl, req.user.user_id) + 1 ? true : false;
```

Comment:
I might have done `return !!(_.indexOf(acl, '*') + 1 || _.indexOf(acl, req.user.user_id) + 1)`. not at all important though